### DISCUSSION TITLE
Enhance Tag Sources functionality to ease tagging an album using track based metadata from other albums

### DISCUSSION THREAD

**Discussion by audiomuze** [puddletag/puddletag#607] [Ideas]
At present you're able to tag a track in tag sources using metadata from other albums by loading that album in tag sources and picking off the tracks you want to take metadata from (because the album you're tagging doesn't exist in a tag sources, but other albums with one or more of the tracks do).  Very useful if you're tagging live albums, soundboards etc. or albums that have not yet been added to any of the tag sources.

This screenshot serves as reference:
![image](https://user-images.githubusercontent.com/6852511/90973863-51ac8480-e526-11ea-9ef4-a7cf00cfe5c3.png)

As can be seen from the above one has to manually/visually search the tag sources track listing looking for matches.

There are a few enhancements that can be made to tag sources to significantly cut down the time and effort required to tag tracks in this manner.

1. Add a Sort button to Tag Sources to toggle between sorting tracks by title and order retrieved from tag source (makes finding matching tracks a lot simpler)
2. Add a toggle to auto match by track title to have puddletag automatically match selected tracks in the editing window with corresponding tracks in the tag source (much like masstagging does, but without regard for tracks count and sequence)
3. Add a text filter that filters the listed tracks by matching the the string a user enters to any part of the tag source's track titles.  

The abovementioned would make short work of using metadata from other albums to populate metadata and enables finding a title quickly when trying to match metadata from one album and applying it to the same tracks in another.

**Comment by sandrotosi** [comment]
I'd be happy to review a PR implementing these new features
